src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyScan.java

```diff
@@ -0,0 +1,292 @@
+
+  /**
+   * A list of configurations associated to OWASP Dependency check plugin which can fail the build.
+   */
+  private static final Map<String, String> FAIL_BUILD_CONFIGURATIONS = new HashMap<>();
+
+  static {
+    FAIL_BUILD_CONFIGURATIONS.put("junitFailOnCVSS", "number"); 
+    FAIL_BUILD_CONFIGURATIONS.put("failBuildOnCVSS", "number");
+    FAIL_BUILD_CONFIGURATIONS.put("failBuildOnAnyVulnerability", "boolean");
+  }
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param fetcher An interface to GitHub.
+   */
+  public UsesOwaspDependencyScan(GitHubDataFetcher fetcher) {
+    super(fetcher);
+  }
+
+  @Override
+  protected Set<Feature> supportedFeatures() {
+    return features();
+  }
+
+  @Override
+  protected ValueSet fetchValuesFor(GitHubProject project) throws IOException {
+    logger.info("Figuring out if the project uses OWASP Dependency Check ...");
+    LocalRepository repository = GitHubDataFetcher.localRepositoryFor(project);
+
+    ValueSet values = new ValueHashSet();
+    checkMaven(repository, values);
+    checkGradle(repository, values);
+
+    return values;
+  }
+
+  /**
+   * Returns the set of OWASP Dependency Check features.
+   */
+  static Set<Feature> features() {
+    return setOf(
+        OWASP_DEPENDENCY_CHECK_USAGE, 
+        OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD);
+  }
+
+  /**
+   * Checks if a project uses OWASP Dependency Check Maven plugin.
+   *
+   * @param repository The project's repository.
+   * @param values set of type {@link ValueSet}.
+   * @return True if the project uses the plugin, false otherwise.
+   */
+  private static void checkMaven(LocalRepository repository, ValueSet values) throws IOException {
+    Optional<InputStream> content = repository.read("pom.xml");
+
+    if (content.isPresent()) {
+      try (InputStream is = content.get()) {
+        Model model = readModel(is);
+        update(values, browse(model, withVisitor()).usageMap);
+      }
+    }
+  }
+
+  /**
+   * Checks if a project uses OWASP Dependency Check Gradle plugin.
+   *
+   * @param repository The project's repository.
+   * @param values set of type {@link ValueSet}.
+   * @return True if the project uses the plugin, false otherwise.
+   */
+  private static void checkGradle(LocalRepository repository, ValueSet values) throws IOException {
+    // Nothing to be done.
+  }
+
+  /**
+   * Update the features with the collected values.
+   *
+   * @param values of type {@link ValueSet}.
+   * @param usageMap a map containing all possible usages and its associated CVSS thresholds.
+   */
+  private static void update(ValueSet values, Map<OwaspDependencyCheckUsage, Double> usageMap) {
+    OwaspDependencyCheckUsage usage = NOT_USED;
+
+    if (usageMap.containsKey(MANDATORY)) {
+      usage = MANDATORY;
+    } else if (usageMap.containsKey(OPTIONAL)) {
+      usage = OPTIONAL;
+    }
+
+    values.update(OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD.value(usageMap.get(usage)));
```

Comment:
Yes, I have a test for this.

I assumed this would happen as `usageMap.put()` is used only if a plugin is found. Hence, this way.

```
if (isDependencyCheck(plugin)) {
        usageMap.put(usageFrom(locations), cvssScoreFrom(plugin.getConfiguration()));
 }
``` 
`OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD.value()` will not throw NPE as, I am doing the null check before I set the value in `OwaspDependencyCheckCvssThreshold#value(Double n)`. Here, if `n` is `null`, then set the `JUNK_VALUE`.

Reply:
I see. I am not sure it's worth it to have this check in `value()`. Please see my comments below.